### PR TITLE
Improve Dispatch Experience

### PR DESCRIPTION
**Pull request by patrick-du**
[Notion ticket](https://www.notion.so/uwblueprintexecs/Dispatch-experience-7a00c9866f244811a4993e4b79fd7583)

## Changes
- Implemented the functionality to update patient run number within the patient modal

## Testing
- Head to an event with a CCP that consists of a patient
- Click on a patient and update the `runNumber`
- Ensure that the button at the bottom right says `Edit` if there is no difference between the old run number and current input, otherwise, it should be `Save Changes`
- Click the `Save Changes` button

- Ensure that mutation occurred by heading to your GraphQL playground and querying for the patient you updated

before opening PR
- [X] check notion ticket
- [X] run linter
- [X] go through file diff

filling out PR
- [X] descriptive title
- [X] update `notion ticket` link
- [X] fill out `Changes`
- [ ] \(optional) add inline comments in diff
- [X] fill out `Testing`
- [ ] \(optional) add `Screenshots`
- [X] assign yourself to the PR

after opening PR
- [X] link PR to notion ticket
- [X] move ticket to `In PR`
- [X] make sure build passes
- [X] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/components/CCPDashboard/PatientInfoTable.tsx [hunk 1]
```diff
@@ -155,9 +161,22 @@ export const PatientInfoTable = ({
...
+  const { enqueueSnackbar } = useSnackbar();
 
-  const [editPatient] = useMutation(EDIT_PATIENT);
+  const [editPatient] = useMutation(EDIT_PATIENT, {
+    update(cache) {
+      const patientId = ((selectedPatient as unknown) as Patient).id;
+      const { patients } = cache.readQuery<any>({
+        query: GET_ALL_PATIENTS,
+      });
+      setSelectedPatient(patients.find((x) => x.id === patientId));
```
Comment (patrick-du) [inline src/components/CCPDashboard/PatientInfoTable.tsx:176]: This ensures that `selectedPatient` always contains a fresh state even when the modal is not closed and reopened. 

Without it, the `handleRunNumber` function will utilize the previous `runNumber` for comparison since `selectedPatient` is only set when the user opens a modal.

File: src/components/CCPDashboard/PatientInfoTable.tsx [hunk 2]
```diff
@@ -155,9 +161,22 @@ export const PatientInfoTable = ({
     patients.find((x) => x.id === patientId)
   );
   const [anchorEl, setAnchorEl] = React.useState(null);
+  const [runNumber, setRunNumber] = React.useState<number | null>(
+    selectedPatient ? selectedPatient.runNumber : null
+  );
+  const [isNewRunNumber, setIsNewRunNumber] = React.useState<boolean>(false);
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientInfoTable.tsx]: is there a way to eliminate this variable? we already have `runNumber`, can't we just check that wherever we currently look at `isNewRunNumber` (i.e. just compare the saved run number with the one that's currently in the input field)?

File: src/components/CCPDashboard/PatientInfoTable.tsx (hunk 1 above)
Comment (petorsong) [reply src/components/CCPDashboard/PatientInfoTable.tsx:176]: just to double check, does this introduce any additional network requests by any chance? i checked my network tab and it seemed reasonable

**Review by petorsong — CHANGES_REQUESTED** [review]
for next time, a screenshot or a short gif would be helpful for reviewers to understand what to expect

added a couple of comments but otherwise looks fine to me!

File: src/components/CCPDashboard/PatientInfoTable.tsx (hunk 1 above)
Comment (patrick-du) [reply src/components/CCPDashboard/PatientInfoTable.tsx:176]: Nope - I don't believe so, according to this https://www.apollographql.com/docs/react/caching/cache-interaction/#readquery
image

File: src/components/CCPDashboard/PatientInfoTable.tsx (hunk 2 above)
Comment (patrick-du) [reply src/components/CCPDashboard/PatientInfoTable.tsx]: Yeah, that makes sense. Thanks, I've made the change in the latest commit.

**Review by petorsong — APPROVED** [review]
🏎️

**Comment by jesskeepswimming** [comment]
image
image
image

All non-edge cases look good-- however, I noticed that changing the run number to an edge case such as 76-54 will show that changes are saved when the run number actually get clears. I'm not sure if this edge case will realistically get encountered though!

**Review by jesskeepswimming — APPROVED** [review]
image
image
image

All non-edge cases look good-- however, I noticed that changing the run number to an edge case such as 76-54 will show that changes are saved when the run number actually get clears. I'm not sure if this edge case will realistically get encountered though!

**Comment by petorsong** [comment]
@jesskeepswimming run numbers will always be numeric - no need to test inputs that contain other characters
